### PR TITLE
Initial ContainsSequence() proposal.

### PR DESCRIPTION
**Pull request by dekimir**
<code>container.ContainsSequence(e1, e2, e3)</code> returns true iff <code>container</code> contains among its elements the sequence <code>{e1, e2, e3}</code>.

TODO: test description and explanation
TODO: test stream containers
TODO: notify listener

### CODE REVIEW

File: googlemock/include/gmock/gmock-matchers.h [hunk 1]
```diff
@@ -2897,6 +2897,112 @@ class EachMatcherImpl : public QuantifierMatcherImpl<Container> {
...
+    StlContainerReference stl_container = View::ConstReference(container);
+    // Tracks positions that match the container so far.
+    ::std::vector<SeqPos> ongoing_matches;
+    for (typename StlContainer::const_iterator current = stl_container.begin();
+         current != stl_container.end(); ++current) {
+      // Assume current begins an ongoing match at position 0.
+      ongoing_matches.push_back(sequence_.begin());
+      // Terminate any ongoing matches that don't extend to current.
+      ongoing_matches.erase(
+          remove_if(ongoing_matches.begin(),
```
Comment (BillyDonahue) [inline googlemock/include/gmock/gmock-matchers.h]: std::remove_if

File: googlemock/include/gmock/gmock-matchers.h [hunk 2]
```diff
@@ -4368,7 +4500,6 @@ template <typename... Args>
 inline internal::AnyOfMatcher<Args...> AnyOf(const Args&... matchers) {
   return internal::AnyOfMatcher<Args...>(matchers...);
 }
-
```
Comment (BillyDonahue) [inline googlemock/include/gmock/gmock-matchers.h]: please revert.

Comment (dmircevski) [inline googlemock/include/gmock/gmock-matchers.h]: Done.

File: googlemock/include/gmock/gmock-matchers.h (hunk 1 above)
Comment (dmircevski) [inline googlemock/include/gmock/gmock-matchers.h]: Done.

File: googlemock/include/gmock/gmock-generated-matchers.h [hunk 3]
```diff
@@ -994,6 +994,211 @@ UnorderedElementsAre(const T1& e1, const T2& e2, const T3& e3, const T4& e4,
...
+    typename internal::DecayArray<T2>::type,
+    typename internal::DecayArray<T3>::type,
+    typename internal::DecayArray<T4>::type,
+    typename internal::DecayArray<T5>::type,
+    typename internal::DecayArray<T6>::type,
+    typename internal::DecayArray<T7>::type,
+    typename internal::DecayArray<T8>::type,
+    typename internal::DecayArray<T9>::type,
+    typename internal::DecayArray<T10>::type> Args;
+  return internal::ContainsSequenceMatcher<Args>(
```
Comment (BillyDonahue) [inline googlemock/include/gmock/gmock-generated-matchers.h]: All of the variadic-simulation here comes from the autogeneration from a .pump script.
To do this properly, you'd need to change the .h.pump file, regenerate the .h from it, and then make a PR of that change.

File: googlemock/include/gmock/gmock-generated-matchers.h [hunk 4]
```diff
@@ -994,6 +994,211 @@ UnorderedElementsAre(const T1& e1, const T2& e2, const T3& e3, const T4& e4,
       e6, e7, e8, e9, e10));
 }
 
+template <typename T1>
+inline internal::ContainsSequenceMatcher<
+    ::testing::tuple<typename internal::DecayArray<T1>::type> >
+ContainsSequence(const T1& e1) {
```
Comment (BillyDonahue) [inline googlemock/include/gmock/gmock-generated-matchers.h]: We'd almost immediately need a "ContainsSequenceArray" (bikeshedded name), to take more than 10 matchers, or a dynamic collection of matchers. Add that?  If you had that, you might not need any of these overloads, really.

File: googlemock/include/gmock/gmock-generated-matchers.h (hunk 3 above)
Comment (dmircevski) [inline googlemock/include/gmock/gmock-generated-matchers.h]: Abandoned autogeneration, as per the next suggestion.

File: googlemock/include/gmock/gmock-generated-matchers.h (hunk 4 above)
Comment (dmircevski) [inline googlemock/include/gmock/gmock-generated-matchers.h]: Done.

**Comment by BillyDonahue** [comment]
Sorry, I said "almost". You still DO need the tuple initializer though. Otherwise the heterogeneous matchers won't have a deducible T for the std::initializer_list<T>, etc.

**Comment by dmircevski** [comment]
Re-added the tuple initializers.

**Comment by dmircevski** [comment]
Ping...

**Comment by gennadiycivil** [comment]
Travis CI build failed. Doing some cleanup and closing outstanding PRs that 
1) don't have all checks OK
2) have been around for a while with no activity

**Pull request closed without merging**